com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/detect/ApplicationMtaMetadataParser.java

```diff
@@ -31,28 +34,38 @@ public static ApplicationMtaMetadata parseAppMetadata(CloudApplication app) {
 
     private static ApplicationMtaMetadata attemptToParseAppMetadata(CloudApplication app) {
         Map<String, String> appEnv = app.getEnvAsMap();
-        DeployedMtaMetadata mtaMetadata = parseMtaMetadata(appEnv);
+        DeployedMtaMetadata mtaMetadata = parseMtaMetadata(app, appEnv);

```

Comment:
Why not pass only the app reference without the appEnv map?

Reply:
getEnvAsMap() is a rather expensive operation, which is why it's extracted at the top of the method.